static/js/compose.js

```diff
@@ -651,6 +651,20 @@ exports.respond_to_message = function (opts) {
     } else {
         msg_type = message.type;
     }
+    // for sending message from preview on clicking enter, if the checkbox is checked.
+    if ($("#new_message_content")[0].style.display === "none") {
+        $("#new_message_content").show();
+        $("#undo_markdown_preview").hide();
+        $("#preview_message_area").hide();
+        $("#markdown_preview").show();
+        // if the checkbox is not checked, option to edit the message will be enabled.
+        if (! $("#enter_sends").is(":checked")) {
+            $("#new_message_content").focus();
+        } else {
+            $("#compose-send-button").click();
```

Comment:
@timabbott How do you feel about simulating clicks?  In some ways it's fine, but I also think it might make more sense to just extract the click handler stuff into a callable function that we call here.